src/vos/vos_io.c

```diff
@@ -2319,6 +2319,15 @@ vos_update_end(daos_handle_t ioh, uint32_t pm_ver, daos_key_t *dkey, int err,
 		}
 	}
 
+	if (err == 0 && ioc->ic_epr.epr_hi > ioc->ic_obj->obj_df->vo_max_write) {
+		if (DAOS_ON_VALGRIND)
+			err = umem_tx_xadd_ptr(umem, &ioc->ic_obj->obj_df->vo_max_write,
+					       sizeof(ioc->ic_obj->obj_df->vo_max_write),
+					       POBJ_XADD_NO_SNAPSHOT);
+		if (err == 0)
+			ioc->ic_obj->obj_df->vo_max_write = ioc->ic_epr.epr_hi;
```

Comment:
I'm not sure how will application use this 'last_write', can it tolerate the inconsistency of 'last_write epoch' and 'last write' caused by PMDK transaction abort (or reboot before vo_max_write being flushed)?

Reply:
we just need an approximation.  It's OK for the use case (described above) to have some false positives.

I removed the snapshot here because it doesn't really matter so long as it's not total garbage (e.g. garbage value rather than an actual timestamp we wrote).

Reply:
I see, thanks.